### DISCUSSION TITLE
Instant loading state and generateMetadata

### DISCUSSION THREAD

**Discussion by mike-tang** [vercel/next.js#50104] [App Router]
Is there a way to produce an instant loading state while generateMetadata is fetching data?

I'm currently using generateMetadata to fetch data for the opengraph title/description on a user profile page. According to the [docs](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#seo), Next.js waits for data fetching inside generateMetadata to complete before streaming UI to the client, but this seems to prevent users from seeing an instant loading state whenever a page is first loaded or [<Link\>](https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#the-link-component) is clicked.

Ideally, I'd like to have some type of "Loading..." indicator in my browser page title while the UI displays an instant loading state (using next app router). 

Any guidance on this is much appreciated – thanks!

From [Next.js App Router docs](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#seo):
> Next.js will wait for data fetching inside [generateMetadata](https://nextjs.org/docs/app/api-reference/functions/generate-metadata) to complete before streaming UI to the client. This guarantees the first part of a streamed response includes <head> tags.

**Comment by CapitaineToinon** [comment]
I'm having the same problem. It doesn't seems to be possible, which is unfortunate. I get what it's trying to do, but in my opinion, if there is a loading.jsx file present the metadata should be inserted later. Having a loading.jsx already hurts SEO so we should at least have the option to enable the same behavior for the metadata.

**Comment by l-you** [comment]
We are waiting for solution.  
There was a long discussion about similar problem. 
https://github.com/vercel/next.js/issues/43548

**Comment by p00000001** [comment]
Open issue on this problem at https://github.com/vercel/next.js/issues/45418

**Comment by rvk220** [comment]
Thanks for raising the issue. Still no updates? 🤔

**Comment by imrashidov** [comment]
has any update? xd
